esiclient/v1/switch.py

```diff
@@ -68,6 +80,47 @@ def take_action(self, parsed_args):
         return ["VLAN", "Ports"], data
 
 
+class ListSwitchPort(command.Lister):
+    """List Switch Ports"""
+
+    log = logging.getLogger(__name__ + ".ListSwitchPort")
+
+    def get_parser(self, prog_name):
+        parser = super(ListSwitchPort, self).get_parser(prog_name)
+        parser.add_argument(
+            "switch",
+            metavar="<switch>",
+            help=_("Switch"))
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        switch = parsed_args.switch
+
+        ironic_client = self.app.client_manager.baremetal
+        neutron_client = self.app.client_manager.network
+
+        ports = list((port for port in ironic_client.port.list(detail=True)
+                      if port.local_link_connection.get(
+                              'switch_info') == switch and
+                      port.internal_info.get('tenant_vif_port_id')))
+        neutron_ports = list(neutron_client.ports())
+
+        data = []
+        for port in ports:
+            switchport = port.local_link_connection.get('port_id')
+            np_id = port.internal_info.get('tenant_vif_port_id')
+            np = next((np for np in neutron_ports if np.id == np_id), None)
+            if np:
+                network_names, _, _ = utils.get_full_network_info_from_port(
+                    np, neutron_client)
+                data.append([switchport, "\n".join(network_names)])
+            else:
+                data.append([switchport, ''])
+        return ["Port", "VLANs"], data
```

Comment:
Should it be ["Port", "networks"]?

Reply:
The network actually show the VLAN, so I'm calling it "VLAN" since that's more switch terminology. I think a subsequent PR might pull apart the network names into separate names/VLAN IDs and let the code decide how to display them.

Reply:
Sounds good!